### PR TITLE
Coefficient Embedding for PolyOverZq and PolyOverQ

### PR DESCRIPTION
**Pull request by Marvin-Beckmann**
**Description**

This PR implements...
- [x] the coefficient embedding for PolyOverQ
- [x] the coefficient embedding for PolyOverZq

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/poly_over_zq/coefficient_embedding.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        FromCoefficientEmbedding, GetCoefficient, GetEntry, GetNumRows, IntoCoefficientEmbedding,
+        SetCoefficient, SetEntry,
+    },
+};
+
+impl IntoCoefficientEmbedding<MatZq> for &PolyOverZq {
+    /// Computes the coefficient embedding of the polynomial
+    /// in a [`MatZq`] as a column vector, where the i-th entry
+    /// of the vector corresponds to the i-th coefficient.
+    /// It inverses the operation of [`PolyOverZq::from_coefficient_embedding`].
```
Comment (Marcel583) [inline src/integer_mod_q/poly_over_zq/coefficient_embedding.rs:25]:
```suggestion
    /// It inverts the operation of [`PolyOverZq::from_coefficient_embedding`].
```

File: src/integer_mod_q/poly_over_zq/coefficient_embedding.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    ///
+    /// Parameters:
+    /// - `size`: determines the number of rows of the embedding. It has to be larger
+    /// than the degree of the polynomial.
+    ///
+    /// Returns a coefficient embedding as a vector if `size` is large enough.
```
Comment (Marcel583) [inline src/integer_mod_q/poly_over_zq/coefficient_embedding.rs:31]:
```suggestion
    /// Returns a coefficient embedding as a column vector if `size` is large enough.
```

File: src/integer_mod_q/poly_over_zq/coefficient_embedding.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    /// not all coefficients can be embedded.
+    fn into_coefficient_embedding(self, size: impl Into<i64>) -> MatZq {
+        let size = size.into();
+        let length = self.get_degree() + 1;
+        assert!(
+            size >= length,
+            "The polynomial can not be embedded in the vector, \
+            as the length of the polynomial ({length}) is larger than \
+            the provided size ({size})."
+        );
```
Comment (Marcel583) [inline src/integer_mod_q/poly_over_zq/coefficient_embedding.rs:58]: panic or write into refactoring

File: src/rational/poly_over_q/coefficient_embedding.rs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        FromCoefficientEmbedding, GetCoefficient, GetEntry, GetNumRows, IntoCoefficientEmbedding,
+        SetCoefficient, SetEntry,
+    },
+};
+
+impl IntoCoefficientEmbedding<MatQ> for &PolyOverQ {
+    /// Computes the coefficient embedding of the polynomial
+    /// in a [`MatQ`] as a column vector, where the i-th entry
+    /// of the vector corresponds to the i-th coefficient.
+    /// It inverses the operation of [`PolyOverQ::from_coefficient_embedding`].
```
Comment (Marcel583) [inline src/rational/poly_over_q/coefficient_embedding.rs:24]: s.o. and others

**Review by Marcel583 — APPROVED** [review]

File: src/integer_mod_q/poly_over_zq/coefficient_embedding.rs (hunk 3 above)
Comment (Marvin-Beckmann) [reply src/integer_mod_q/poly_over_zq/coefficient_embedding.rs:58]: added to refactoring

**Pull request merged**